pkg/segment/writer/segwriter.go

```diff
@@ -561,37 +544,69 @@ func getFinalBaseSegDirFromActive(activeBaseSegDir string) (string, error) {
 }
 
 /*
-Adds the fullWord and sub-words to the bloom
+Adds the fullWord and sub-words (lowercase as well) to the bloom
 Subwords are gotten by splitting the fullWord by whitespace
+NOTE: This function may modify the incoming byte slice
 */
-func addToBlockBloom(blockBloom *bloom.BloomFilter, fullWord []byte) uint32 {
+func addToBlockBloomBothCases(blockBloom *bloom.BloomFilter, fullWord []byte) uint32 {
 
 	var blockWordCount uint32 = 0
 	copy := fullWord[:]
 
+	// we will add the lowercase to bloom only if there was an upperCase and we
+	// had to convert
+	hasUpper := utils.HasUpper(copy)
+
+	// add the original full
 	if !blockBloom.TestAndAdd(copy) {
 		blockWordCount += 1
 	}
 
-	var foundWord bool
+	var hasSubWords bool
 	for {
 		i := bytes.Index(copy, BYTE_SPACE)
 		if i == -1 {
 			break
 		}
-		foundWord = true
+		hasSubWords = true
+		// add original sub word
 		if !blockBloom.TestAndAdd(copy[:i]) {
 			blockWordCount += 1
 		}
+
+		// add original sub word lowercase
+		if hasUpper {
+			if !blockBloom.TestAndAdd(utils.BytesToLowerInPlace(copy[:i])) {
+				blockWordCount += 1
+			}
+		}
 		copy = copy[i+BYTE_SPACE_LEN:]
 	}
 
 	// handle last word. If no word was found, then we have already added the full word
-	if foundWord && len(copy) > 0 {
+	if hasSubWords && len(copy) > 0 {
 		if !blockBloom.TestAndAdd(copy) {
 			blockWordCount += 1
 		}
+		if !blockBloom.TestAndAdd(utils.BytesToLowerInPlace(copy)) {
+			blockWordCount += 1
+		}
 	}
+
+	if hasUpper {
+		if hasSubWords {
+			// if subwords, then add the original full's lowercase
+			// but have to use the original var, since the copy is pointing to last subword
+			if !blockBloom.TestAndAdd(utils.BytesToLowerInPlace(fullWord[:])) {
+				blockWordCount += 1
+			}
+		} else {
+			if !blockBloom.TestAndAdd(utils.BytesToLowerInPlace(copy)) {
+				blockWordCount += 1
+			}
+		}
```

Comment:
nit: can't the `hasSubwords` and `!hasSubwords` cases be handled the same way (using `fullWord`)?

Reply:
yup, good point. Fixed it